lxd/certificates.go

```diff
@@ -614,20 +614,12 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 		}
 
 		cert = r.TLS.PeerCertificates[len(r.TLS.PeerCertificates)-1]
-		networkCert := s.Endpoints.NetworkCert()
-		if networkCert.CA() != nil {
-			// If we are in CA mode, we only allow adding certificates that are signed by the CA.
-			trusted, _, _ := util.CheckCASignature(*cert, networkCert)
-			if !trusted {
-				return response.Forbidden(fmt.Errorf("The certificate is not trusted by the CA or has been revoked"))
-			}
-		}
 	} else {
 		return response.BadRequest(fmt.Errorf("Can't use TLS data on non-TLS link"))
 	}
 
 	// Check validity.
-	err = certificateValidate(cert)
+	err = certificateValidate(d.endpoints.NetworkCert(), cert)
```

Comment:
can we avoid calling d.endpoints.NetworkCert multiple times in same function (it takes a lock you see).